src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -838,7 +838,7 @@ def forward(
             attention_mask is not None
             and hasattr(self.config, "_flash_attn_2_enabled")
             and self.config._flash_attn_2_enabled
-            and past_key_values is not None
+            and use_cache
```

Comment:
In the first autoregressive pass, `past_key_values` is None. In the following passes, we may have masks as

```
[[1, 1, 0, 0, 1],
[1, 1, 1, 1, 1]]
```

and `is_padding_right` is wrongfully evaluated to False.